common/djangoapps/track/eventshim.py

```diff
@@ -0,0 +1,380 @@
+"""
+EventShims that create
+"""
+
+# pylint: disable=missing-docstring
+
+import json
+import logging
+
+from opaque_keys import InvalidKeyError
+from opaque_keys.edx.keys import UsageKey
+
+log = logging.getLogger(__name__)
+
+
+class DottedPathMapping(object):
+    """
+    Dictionary-like object for creating keys of dotted paths.
+
+    If a key is created that ends with a dot, it will be treated as a path
+    prefix.  Any value whose prefix matches the dotted path can be used
+    as a key for that value, but only the most specific match will
+    be used.
+    """
+
+    def __init__(self, registry=None):
+        self._match_registry = {}
+        self._prefix_registry = {}
```

Comment:
`2`: This makes usability worse for the programmer, because the registration then has to happen in a different place from the definition of the transformer.  This could be worth it, if we actually have a need for the full power of regexes, but I don't see that we do.

`4`: How do you constrain the regex?  In order to do this sensibly, it seems like we already have to define a subset of regular expressions that we want to support, and then we haven't solved the documentation issue.  And if we don't constrain it, we end up with a system that is hard to test.  How do you make sure that your transformer is handling the right events, and only the right events? 

Another (somewhat minor) problem with a regex-based syntax is that `.` has a special meaning, so the simple case (of exact matches) already has poor ergonomics (`r'edx\.ui\.lms\.sequence\.next_selected'`).

The current version, while slightly non-standard, is readable and simple. It solves the problem we have, without introducing extra complexity.  If there's a need down the line to handle arbitrary regular expressions, that would be easy to refactor this, but right now, it feels like it solves trivial problems that don't need to be solved, while introducing bigger ones.